sinks/signalfx/signalfx_test.go

```diff
@@ -713,13 +709,14 @@ func TestSignalFxClientByTagUpdater(t *testing.T) {
 	derived := newDerivedProcessor()
 	perBatch := 1
 
-	sink, err := NewSignalFxSink("host", "glooblestoots", map[string]string{"yay": "pie"}, logrus.New(), fallback, "test_by", map[string]DPClient{}, nil, nil, derived, perBatch, "", true, 5*time.Millisecond, "", server.URL, server.Client())
+	sink, err := NewSignalFxSink("host", "glooblestoots", map[string]string{"yay": "pie"}, logrus.New(), fallback, "test_by", map[string]DPClient{}, nil, nil, derived, perBatch, "", true, dynamicKeyRefreshPeriod, "", server.URL, server.Client())
```

Comment:
One thing that occurred to me: Instead of passing in a Duration, can we pass in a `<-chan Time`, as from a Ticker? That way we could trigger the refresh manually in a test (and making this 100% deterministic would only require that we hook back the "we did the refresh" signal).

Reply:
We could do that, yes! It'd require a bit more work elsewhere. I'd prefer to ticket that for a future fix, because this flaky test is currently blocking other PRs that we'd like to merge, but I think that's a good idea.